### DISCUSSION TITLE
Playing .mkv, .avi, etc. videos?

### DISCUSSION THREAD

**Discussion by docyx** [tauri-apps/tauri#4815] [Q&A]
My app needs the ability to play .mkv, .avi and other types of media formats not usually supported by web browsers.

I'm a bit new when it comes to Rust -- don't think I'm up for building a transcoder just yet, so I just thought I would ask if there are any options. Thanks

**Comment by khalilyfz** [comment]
build a transcoder...

**Comment by CarbonPool** [comment]
Use ffmpeg to transcode and push streams

**Comment by CaseSens** [comment]
One solution would be to use a publicly available rust crate like [ffmpeg-next](https://crates.io/crates/ffmpeg-next), however this can come with it's own set of problems like ffmpeg being a nightmare to compile during builds. I think some people have been able to statically compile ffmpeg directly into their app, but I was not able to get it to work.

A simpler solution may very well be to bundle ffmpeg binary with your app, and call it during runtime using `std::process::Command("ffmpeg")`. Keep in mind this is more of a band-aid solution.
